src/plugins/git/porcelain.js

```diff
@@ -0,0 +1,264 @@
+    return this.graph.node(this.nodeAddress);
+  }
+
+  address(): Address {
+    return this.nodeAddress;
+  }
+}
+
+export class Commit extends BaseGitPorcelain {
+  static from(n: BaseGitPorcelain): Commit {
+    if (n.type() !== "COMMIT") {
+      throw new Error(`Unable to cast ${n.type()} to Commit`);
+    }
+    return new Commit(n.graph, n.nodeAddress);
+  }
+
+  hash(): Hash {
+    return this.address().id;
+  }
+
+  parents(): Commit[] {
+    return this.graph
+      .neighborhood(this.nodeAddress, {
+        nodeType: "COMMIT",
+        edgeType: "HAS_PARENT",
+        direction: "OUT",
+      })
+      .map(({neighbor}) => new Commit(this.graph, neighbor));
+  }
+
+  tree(): Tree {
+    const trees = this.graph
+      .neighborhood(this.nodeAddress, {
+        nodeType: "TREE",
+        edgeType: "HAS_TREE",
+        direction: "OUT",
+      })
+      .map(({neighbor}) => new Tree(this.graph, neighbor));
+    if (trees.length !== 1) {
+      throw new Error(
+        `Commit ${stringify(this.nodeAddress)} has wrong number of trees`
+      );
+    }
+    return trees[0];
+  }
+}
+
+export class Tree extends BaseGitPorcelain {
+  static from(n: BaseGitPorcelain): Tree {
+    if (n.type() !== "TREE") {
+      throw new Error(`Unable to cast ${n.type()} to Tree`);
+    }
+    return new Tree(n.graph, n.nodeAddress);
+  }
+
+  hash(): Hash {
+    return this.address().id;
+  }
+
+  entries(): TreeEntry[] {
+    return this.graph
+      .neighborhood(this.nodeAddress, {
+        nodeType: "TREE_ENTRY",
+        edgeType: "INCLUDES",
+        direction: "OUT",
+      })
+      .map(({neighbor}) => new TreeEntry(this.graph, neighbor));
+  }
+
+  entry(name: string): ?TreeEntry {
+    return this.entries().filter((te) => te.name() === name)[0];
+  }
+}
+
+export class TreeEntry extends BaseGitPorcelain {
+  static from(n: BaseGitPorcelain): TreeEntry {
+    if (n.type() !== "TREE_ENTRY") {
+      throw new Error(`Unable to cast ${n.type()} to TreeEntry`);
+    }
+    return new TreeEntry(n.graph, n.nodeAddress);
+  }
+
+  name(): string {
+    const node: Node<TreeEntryNodePayload> = (this.node(): any);
+    return node.payload.name;
+  }
+
+  evolvesTo(): TreeEntry[] {
+    return this.graph
+      .neighborhood(this.nodeAddress, {
+        nodeType: "TREE_ENTRY",
+        edgeType: "BECOMES",
+        direction: "OUT",
+      })
+      .map(({neighbor}) => new TreeEntry(this.graph, neighbor));
+  }
+
+  evolvesFrom(): TreeEntry[] {
+    return this.graph
+      .neighborhood(this.nodeAddress, {
+        nodeType: "TREE_ENTRY",
+        edgeType: "BECOMES",
+        direction: "IN",
+      })
+      .map(({neighbor}) => new TreeEntry(this.graph, neighbor));
+  }
+
+  contents(): Tree | Blob | SubmoduleCommit {
+    const neighbors = this.graph
+      .neighborhood(this.nodeAddress, {
+        edgeType: "HAS_CONTENTS",
+        direction: "OUT",
+      })
+      .map(({neighbor}) => neighbor);
+    if (neighbors.length !== 1) {
```

Comment:
I believe that this invariant does not hold for submodule commit nodes. See:
https://github.com/sourcecred/sourcecred/blob/115d7f3921c5145180b472f1678edaf804d0973f/src/plugins/git/createGraph.js#L153-L198

We might want to add such a case to our example-git repository. (Later.)